src/Api/StopRecurrentPaymentApiHandler.php

```diff
@@ -82,6 +82,15 @@ public function handle(array $params): ApiResponseInterface
             return $response;
         }
 
+        if (!$this->recurrentPaymentsRepository->canBeStoppedByUser($recurrentPayment)) {
+            $response = new JsonResponse([
+                'message' => "Payment gateway [{$recurrentPayment->payment_gateway->code}] is unstoppable by user.",
+                'code' => 'user_unstoppable_recurrent_payment_gateway'
+            ]);
+            $response->setHttpCode(Response::S409_CONFLICT);
```

Comment:
I probably wouldn't use the HTTP 409 (Conflict), because it would indicate there's something with the actual recurrent payment, not its gateway. Also it's intended to be used in the scenarios, where the conflict is resolvable, which this isn't.

We'll most likely go just with the HTTP 403 (Forbidden)